l10n_do_ncf_validation/models/account_move.py

```diff
@@ -0,0 +1,120 @@
+import requests
+from odoo.tools import safe_eval
+
+from odoo import models, _
+from odoo.exceptions import ValidationError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _has_valid_ncf(self):
+        """
+        Query external service to check NCF status
+        :return: boolean: True if valid NCF, otherwise False
+        """
+        self.ensure_one()
+
+        def check_rnc_format(vat):
+            if not vat or not str(vat).isdigit() or len(vat) not in (9, 11):
+                raise ValidationError(
+                    _("A valid RNC/Cédula is required to request a NCF validation.")
+                )
+
+        rnc = (
+            self.company_id.vat
+            if self.type not in ("in_invoice", "in_refund")
+            else self.partner_id.vat
+        )
+        check_rnc_format(rnc)
+
+        ncf = self.ref
+        if not ncf or not len(ncf) in (11, 13) or ncf[0] not in ("B", "E"):
```

Comment:
```suggestion
        if not ncf or len(ncf) not in (11, 13) or ncf[0] not in ("B", "E"):
```
Me parece que así es mas agradable a la vista. Pero la forma actual igual hace lo que se requiere.